docker_push_latest_if_changed.py

```diff
@@ -1,5 +1,120 @@
 #!/usr/bin/env python3
 import argparse
+import hashlib
+from collections import namedtuple
+
+import docker_util
+from verbose_print import print_v
+from verbose_print import print_vv
+from verbose_print import set_print_level
+
+
+ImageParityFields = namedtuple(
+    'ImageParityFields',
+    ('image_cmds_hash', 'packages_hash'),
+)
+
+GET_PACKAGE_LIST_COMMAND = ['dpkg', '-l']
+
+
+def _docker_push_latest_if_changed(source, target, is_dry_run):
+    _validate_source(source)
+    target = _get_sanitized_target(source, target)
+
+    print_v('Pushing source image')
+    docker_util.push_image(source, is_dry_run)
+    try:
+        print('Pulling target image...')
+        docker_util.pull_image(target)
+    except docker_util.ImageNotFoundError:
+        print((
+            f'Target image {target} was not found in the registry. '
+            'Going to attempt to tag and push the target image anyway.'
+        ))
+        docker_util.tag_image(source, target, is_dry_run)
+        docker_util.push_image(target, is_dry_run)
+    else:
+        if _is_image_changed(source, target):
```

Comment:
might read more naturally as `if _image_changed(...):`

Reply:
I think we still prefer to keep prefixes like if/should/has before boolean parameters.
I'll try renaming to `_has_image_changed(...):`
Maybe that would seem more natural?